### PR TITLE
Next.js 13 support

### PR DESCRIPTION
**Pull request by martpie**
- [x] What breaks?
- [x] Play with the  [new `transpilePackages` option](https://github.com/martpie/next-transpile-modules/issues/280#issuecomment-1292615030)
- [x] What can we remove from this plugin with `transpilePackages`
- [ ] Support `app/`
- [ ] Check Turbopack

---

About the next `transpilePackages` option:

- ✅ `test-local-module` (import and use local ES module)
- ✅ `test-npm-module` (import and use npm ES module)
- ❌ `test-local-typescript-module` (import a local module written in TypeScript)
- ❌ `test-css-module` (import local component importing a css module)
- ❌ `test-scss-module` (import local component importing a scss module)
- ❌ `test-global-ccss` (import local component import css to be injected globally)
- ❌ `test-global-scss` (import local component import scss to be injected globally)

TLDR: Transpiling classic modules is ok. Transpiling local TypeScript libraries or component using CSS does not work.

### CODE REVIEW

**Comment by martpie** [comment]
Quickly checked the new `transpilePackages`, and it's not working on most of the use cases `next-transpile-modules` is trying to solve (css imports in dependencies especially).

Considering the feature is also experimental, I don't foresee `next-transpile-modules` dying anytime soon until those usecases are fixed at Next.js level, and/or a good alternative exists.

I will also open an issue on Next.js' side to explain those different usecases, and see what the team thinks about them.

**Comment by martpie** [comment]
👀
